### PR TITLE
Do not submit saved backups as submissions

### LINKED ISSUE #184: Saved .ok_messages are uploaded with incorrect flags
Saved Ok Messages are uploaded with the submit argument set to `self.args.submit` - which can cause an issue if there are a lot of submissions saved locally. It also means that old code gets flagged as a submissions. 

It results in things like this: (~370 submissions within a minute) 
image

Proposed fix: 
 - Old Messages are never sent with `submit: True`. If the submission failed, don't flag it as a submit (but leave the submit file in messages)

### PR DESCRIPTION
**Pull request by Sumukh**
Resolves #184

### CODE REVIEW

File: client/protocols/backup.py [hunk 1]
```diff
@@ -26,13 +26,24 @@ def run(self, messages):
...
         log.info('Authenticated with access token %s', access_token)
+        log.info('Sending unsent messages %s', access_token)
+
+        # Send submission
+        response = self.send_all_messages(access_token, subm_messages, current=True)
 
-        response = self.send_all_messages(access_token, message_list)
-        prefix='http' if self.args.insecure else 'https'
+        # Send all backups (saved and current)
+        self.send_all_messages(access_token, message_list, current=False)
```
Comment (Sumukh) [inline client/protocols/backup.py]: Oh. I realized this should be also be response. Fixing in a sec.

File: client/protocols/backup.py [hunk 2]
```diff
@@ -26,13 +26,24 @@ def run(self, messages):
...
+            subm_messages = [messages]
+        else:
+            subm_messages = []
 
         access_token = auth.authenticate(False)
         log.info('Authenticated with access token %s', access_token)
+        log.info('Sending unsent messages %s', access_token)
+
+        # Send submission
+        response = self.send_all_messages(access_token, subm_messages, current=True)
```
Comment (Sumukh) [inline client/protocols/backup.py]: In the event of a backup this list is empty and just returns. Going to refactor this now.

File: client/protocols/backup.py [hunk 3]
```diff
@@ -26,13 +27,26 @@ def run(self, messages):
...
         log.info('Authenticated with access token %s', access_token)
+        log.info('Sending unsent messages %s', access_token)
 
-        response = self.send_all_messages(access_token, message_list)
-        prefix='http' if self.args.insecure else 'https'
+        # Messages from the current backup to send first
+        subm_messages = [messages] if self.args.submit else []
+
+        if self.args.submit:
+            response = self.send_all_messages(access_token, subm_messages,
```
Comment (brianhou) [inline client/protocols/backup.py]: Is there a reason this isn't `message_list.insert(0, messages)`? Is the distinction between `current` being `True` or `False` important?

File: client/protocols/backup.py [hunk 4]
```diff
@@ -81,19 +95,25 @@ def dump_unsent_messages(cls, message_list):
             os.fsync(f)
 
 
-    def send_all_messages(self, access_token, message_list):
-        action = 'Submit' if self.args.submit else 'Back up'
+    def send_all_messages(self, access_token, message_list, current=False):
+        if not current or not self.args.submit:
+            action = 'Back up'
```
Comment (brianhou) [inline client/protocols/backup.py]: Should be `Backup` to be consistent with elsewhere.

File: client/protocols/backup.py (hunk 3 above)
Comment (brianhou) [inline client/protocols/backup.py]: Ah, I just got to `send_messages`. So it basically converts all stale submit runs into regular backups?

Comment (Sumukh) [inline client/protocols/backup.py]: Yup.

**Comment by brianhou** [comment]
Looks fine to me. Thanks!

**Pull request merged**